spec/preservation_ingest/complete_ingest_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe Robots::SdrRepo::PreservationIngest::CompleteIngest do
+  let(:bare_druid) { 'jc837rq9922' }
+  let(:druid) { "druid:#{bare_druid}" }
+  let(:deposit_dir_pathname) { Pathname(File.join(File.dirname(__FILE__), '..', 'fixtures', 'deposit', 'complete-ingest')) }
+  let(:deposit_bag_pathname) { Pathname(File.join(deposit_dir_pathname, bare_druid)) }
+  let(:mock_so) { instance_double(Moab::StorageObject, deposit_bag_pathname: deposit_bag_pathname) }
+  let(:this_robot) { described_class.new }
+
+  context '#perform' do
+    before do
+      allow(deposit_bag_pathname).to receive(:rmtree)
+      allow(Moab::StorageServices).to receive(:find_storage_object).and_return(mock_so)
+      FileUtils.mkdir_p(deposit_bag_pathname)
+      FileUtils.touch(deposit_bag_pathname + 'bagit_file.txt')
+    end
+    after do
+      deposit_dir_pathname.rmtree if deposit_dir_pathname.exist?
+    end
+
+    it 'raises ItemError if it fails to remove the deposit bag' do
+      expect(deposit_bag_pathname.exist?).to be true
```

Comment:
Not critical, but you can use the `exist` matcher here, if you like:
```ruby
expect(deposit_bag_pathname).to exist
```

Reply:
true - will file away for future use.  (xxx? predicates can be the expectation in rspec tests - which I knew but wasn't thinking about)